controllers/feeds/detailFeed.js

```diff
@@ -63,6 +63,26 @@ module.exports = async (req, res) => {
 
   newItem.latest_reactions.comment = comments;
 
+  if (Array.isArray(newItem?.latest_reactions?.upvotes)) {
+    const modifiedUpvotes = newItem?.latest_reactions?.upvotes?.reduce((acc, current) => {
+      current.user.data = {};
+      acc.push(current);
+      return acc;
+    }, []);
+
+    newItem.latest_reactions.upvotes = modifiedUpvotes;
+  }
+
+  if (Array.isArray(newItem?.latest_reactions?.downvotes)) {
+    const modifiedDownvotes = newItem?.latest_reactions?.downvotes?.reduce((acc, current) => {
+      current.user.data = {};
+      acc.push(current);
+      return acc;
+    }, []);
+
+    newItem.latest_reactions.downvotes = modifiedDownvotes;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The current implementation of removing user data from `upvotes` and `downvotes` is not optimal. It directly modifies the `current.user.data` object inside the `reduce` function, which can lead to unexpected side effects because it mutates the original array. Instead, you should create a new object for each user without the `data` field.

```diff
-  current.user.data = {};
-  acc.push(current);
+  const newUser = { ...current.user, data: {} };
+  const newVote = { ...current, user: newUser };
+  acc.push(newVote);
```
This way, you're not mutating the original objects but creating new ones with the modified data. This approach is safer and more predictable.


<!-- This is an auto-generated comment by OSS CodeRabbit -->